.binder/requirements.txt

```
@@ -1,7 +1,7 @@
 bqplot
 ipyleaflet
 jupyterlab-myst
-jupyterlab~=3.0
+jupyterlab
```

Does it work if you specifically pin jupyterlab?

```suggestion
jupyterlab==4.0.2
```

---

@martinRenou Honestly, I'm having trouble getting past the yarn 3 requirement.

Even trying _locally_ to build ipywidgets with yarn 3, I can't get past the `yarn install` stage. ...Actually, the whole thing is complicated enough that I'll open a new Issue for it, rather than cluttering this PR.

---

I haven't been able to get past the yarn 3 requirement to even _try_, TBH. Rather than clutter this PR with it, I opened #3797 documenting that issue.

---

(ISTR I tried something like that, and it _did_ work to set the jupyterlab version, but still complained that yarn wasn't version 3. And upgrading yarn to version 3 causes all of #3797 to happen.)

---

So, long story short it turned out my yarn issues were due to my misinterpreting dependency warnings, followed by a silent segfault during the download (Fetch) stage, as dependency **errors**.

I switched to running `yarn` in a Docker container and now it no longer segfaults, so yay for that. Now I'm fighting a bit with `conda` not liking being "activated" from a Dockerfile run, but at least running interactively inside the container all seems well.

---

That, along with a `yarn set version berry` line added to `postBuild` to ensure that yarn is at version 3, seems to have fixed things.